ApiSpecifications.md

```diff
@@ -399,7 +399,15 @@ Status Code: 500
 ## Delete Connection API
 
 This API should clean up a specific connection id in a given org id (workspace_id) and cloud provider.
-This API is currently only used in automated tests.
+This API is used in automation testing and is optional for the Partner Connect experience.
```

Comment:
@cathy-db I have some rough memory that we agreed on making it optional, could you remind me what is the discussion back then? 
Thinking of the feature now, I feel it would be good to mark it required for the future partners. But mentioning it is optional to clean up the resources from the partner side.

Reply:
Basically partner could choose to do nothing when got the API call. But the extra bonus of cleaning up the resources is that the integration will be tested by Databricks.

Reply:
Discussion back then mentioned making it optional to implement and that the API would be tested if they did implement it

Reply:
Thanks for the context. Taking another thought on this, to make it better maintain for the long run, it would be better to require partner providing an API, but don't require them to do the actual deletion. Basically `delete connection api` is required, but the response could be `deleted resources` `pending` or `acknowledged` -- the three response you had in place. How do you feel about it?

Reply:
Sounds good, I'll make a new private PR for the optional -> required doc change